### PR TITLE
feat: Group Conversation - Receipt Mode System Message

### PR DESCRIPTION
**Pull request by alexandreferris**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

[AR-2712 - Show a system message when read receipts are turned on](https://wearezeta.atlassian.net/browse/AR-2712)

This PR is only for handling receipt mode when a new group event is received or in-app user creates a group.
The handling of receipt mode from changes in group settings screen will be handled in a later PR.

### Issues

New group conversations [from event or user in-app created] were not adding a system message for receipt mode.

### Causes (Optional)

It was not implemented.

### Solutions

Add implementation to always add a system message informing the members of the conversation what is the receipt mode status if the creator/receiver of the new group event matches the criteria of:
- isSelfATeamMember
- conversation is of GROUP type
- receipt mode status is NOT NULL (for users without a team it will be null when invoking `CreateGroupConversationUseCase` thus not showing for them)

### Dependencies (Optional)

A PR in AR will be raised upon this one being merged

#### How to Test

Currently the only way to test this in Kalium is to build this branch in AR and create/receive a new group event and then check `MessageNewConversationReceiptModeContent` table values that is contains the selected receipt mode.

### CODE REVIEW

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/conversation/CreateGroupConversationUseCaseTest.kt [hunk 1]
```diff
@@ -137,6 +149,40 @@ class CreateGroupConversationUseCaseTest {
             .wasInvoked(exactly = once)
     }
 
+    @Test
+    fun givenNameMembersAndOptions_whenCreatingGroupConversation_thenPersistSystemMessageForReceiptMode() = runTest {
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/conversation/CreateGroupConversationUseCaseTest.kt:153]: Could you also test for when the user is not a team member, so it doesn't insert the system message?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/CreateGroupConversationUseCase.kt [hunk 2]
```diff
@@ -20,11 +25,15 @@ import com.wire.kalium.util.DateTimeUtil
...
  */
+@Suppress("LongParameterList")
 class CreateGroupConversationUseCase internal constructor(
     private val conversationRepository: ConversationRepository,
     private val conversationGroupRepository: ConversationGroupRepository,
     private val syncManager: SyncManager,
-    private val currentClientIdProvider: CurrentClientIdProvider
+    private val currentClientIdProvider: CurrentClientIdProvider,
+    private val selfUserId: UserId,
+    val persistMessage: PersistMessageUseCase,
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/CreateGroupConversationUseCase.kt:35]:
```suggestion
    private val persistMessage: PersistMessageUseCase,
```

If necessary, please make `PersistMessageUseCase` internal as well.

It shouldn't be public in the first place, and from what I've seen, making it `internal` won't cause any breaking changes

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Great work!

Just a couple of tiny changes.

**Review by vitorhugods — APPROVED** [review]
👨‍🍳🤌

**Review by ohassine — APPROVED** [review]
LGTM 👌 🚀 just before merging, could you please check that AR can be compiled with these changes.

**Comment by alexandreferris** [comment]
> LGTM 👌 🚀 just before merging, could you please check that AR can be compiled with these changes.

AR will break but I already have a PR to be opened and reviewed so it doesn't stay broken for too long

**Pull request merged**